packages/web-pkg/src/composables/actions/files/useFileActionsPaste.ts

```diff
@@ -46,28 +54,37 @@ export const useFileActionsPaste = () => {
     sourceSpace: SpaceResource
     resources: Resource[]
   }) => {
-    const copyMove = new ResourceTransfer(
+    const resourceTransfer = new ResourceTransfer(
       sourceSpace,
       resources,
       targetSpace,
       unref(currentFolder),
       currentFolder,
       clientService,
-      loadingService,
       $gettext,
       $ngettext
     )
-    let movedResourcesPromise: Promise<Resource[]>
-    if (clipboardStore.action === ClipboardActions.Cut) {
-      movedResourcesPromise = copyMove.perform(TransferType.MOVE)
-    }
-    if (clipboardStore.action === ClipboardActions.Copy) {
-      movedResourcesPromise = copyMove.perform(TransferType.COPY)
+
+    const transferData = await resourceTransfer.getTransferData(unref(transferType))
+    if (!transferData.length) {
+      return
     }
-    return movedResourcesPromise.then((movedResources) => {
-      const loadingResources = []
+
+    const originalCurrentFolderId = unref(currentFolder)?.id
+
+    startWorker(transferData, async ({ successful, failed }) => {
+      resourceTransfer.showResultMessage(failed, successful, unref(transferType))
+
```

Comment:
Wouldn't it make sense to have an early return here after the result message if the length of `successful` is `0`?